content/ADR-189-social-service-M2.md

```diff
@@ -0,0 +1,165 @@
+
+## Specification
+
+Due to the benefits explained over the SSE solution, Websocket is the solution that will be further explored in this document.
+
+The new flow for a friendship will be the following:
+
+```mermaid
+sequenceDiagram
+    participant User 1
+    participant User 2
+    participant Social service
+    participant Matrix
+
+    User 1 ->> Social service: Add friend (new friend request)
+    Social service ->>  User 2: New request from user 1
+    User 2 ->> Social service: Accept friendship
+    Social service ->> Matrix: create room
+    Social service -->>+ User 1: New FriendshipId: roomId
+    Social service -->>+ User 2: New FriendshipId: roomId
+```
+
+This new flow implies that rooms in Matrix will not be created until the friendship is established. This does not make the opposite true, if the friendship ends the room won't be destroyed (to prevent the history from being lost).
+
+Now, when a user logs in to decentraland, the friendship requests will also be obtained from the social service instead of the matrix sdk.
+
+First, the client will need to log in via HTTP, then establish a connection with the server, once correctly established, the server will then send two messages one with the current friends and another with the pending friend requests.
+
+```mermaid
+sequenceDiagram
+    participant User 1
+    participant Social service
+    participant Matrix
+
+    User 1 ->>+ Social service: Login
+    Social service ->>+ Matrix: Login
+    Matrix -->>- Social service: bearer token
+    Social service -->>- User 1: bearer token
+    User 1 ->>+ Social service: Connect via Websocket (token)
+    Social service ->> User 1: Friends
+    Social service ->> User 1: Friend Requests
+    loop  Connection messages examples
+    User 1 ->> Social service: Friend event (ACCEPT, DELETE, etc.)
+    Social service ->> User 1: Friend event (ACCEPT, DELETE, etc.)
+    end
+    User 1 ->> Social service: Disconnect WebSocket
+    Social service ->>- User 1: Disconnect WebSocket
+```
+
+## Implementation
+
+### Unity Renderer changes
+
+Given that Browser-interface is going to be slowly migrated into unity's codebase, the implementation of this feature will be done directly on unity's side too. Since this feature includes friends and friend requests, all of the code needed to handle friends and friendship requests will also be moved to unity, decoupling the dependency on Browser-interface. Still, channels, presence, and messaging will stay on the Browser interface side, but within time they will also be migrated to unity and will be able to leverage the new communication system.
+
+To prevent migrating all the code into unity-renderer's code and adding a lot of logic, this ADR proposes to add a new library in C# that will handle the logic around friendships and modify the logic in unity renderer to become more of an orchestrator of events.
```

Comment:
```suggestion
To prevent migrating all of the code into Unity Renderer's code and adding a lot of logic, this ADR proposes adding a new C# library to handle the logic around friendships. The logic in Unity Renderer would be adjusted to focus on event orchestration rather than handling all of the logic directly.
```